### ISSUE TITLE
Regression:  property of complex decorator fails.

### ISSUE THREAD

**Issue by wpk-nist-gov** [microsoft/pyright#7740]
Decorating a class method with `property` that is already decorated by a complicated decorator fails with `pyright==1.1.359`

I opened a similar issue (#7667) that `pyright==1.1.359` addressed, but the below is still failing.

**Code or Screenshots**
The below code works with `pyright<1.1.358`, but fails otherwise.

```python
from typing import ParamSpec, TypeVar, Callable, TypeAlias, Protocol, Concatenate

P = ParamSpec("P")
R = TypeVar("R")
S = TypeVar("S", bound="HasAttr")
Dec: TypeAlias = Callable[Concatenate[S, P], R]


class HasAttr(Protocol):
    my_attr: str

def decorate(func: Dec[S, P, R]) -> Dec[S, P, R]:
    def wrapper(self: S, /, *args: P.args, **kwargs: P.kwargs) -> R:
        print("with attr", self.my_attr)

        return func(self, *args, **kwargs)
    return wrapper


# This would fail because it's not an instance of HasAttr.  This is good.
# class FailOnPurpose:
#     @decorate
#     def meth(self) -> int:
#         return 1

class Works:
    my_attr: str = "Works"

    @decorate
    def meth(self) -> int:
        return 1


    # But this fails
    @property
    @decorate
    def prop(self) -> int:
        return 1


x = Works()
reveal_type(x.meth)
reveal_type(x.meth())
reveal_type(x.prop)
```

With `pyright==1.1.359`:
```bash
/tmp2.py
  /tmp2.py:46:13 - information: Type of "x.meth" is "() -> int"
  /tmp2.py:47:13 - information: Type of "x.meth()" is "int"
  /tmp2.py:49:13 - information: Type of "x.prop" is "Unknown"
  /tmp2.py:49:15 - error: Cannot access attribute "prop" for class "Works"
    Failed to call method "__get__" for descriptor class "property" (reportAttributeAccessIssue)
1 error, 0 warnings, 3 informations
```

With `pyright==1.1.357`:
```bash
/tmp2.py
  /tmp2.py:46:13 - information: Type of "x.meth" is "() -> int"
  /tmp2.py:47:13 - information: Type of "x.meth()" is "int"
  /tmp2.py:49:13 - information: Type of "x.prop" is "int"
0 errors, 0 warnings, 3 informations
```

I use decorators like this to enforce certain functionality. 

**VS Code extension or command-line**
Command line

**Comment by erictraut** [comment]
Note to myself: here's a slightly simplified repro...

It appears that the problem is specifically related to a type alias. It doesn't occur without a type alias.

```python
from typing import Callable, Protocol, Concatenate

type Dec[S: HasAttr, **P, R] = Callable[Concatenate[S, P], R]

class HasAttr(Protocol):
    my_attr: str

def decorate1[S: HasAttr, **P, R](func: Dec[S, P, R]) -> Dec[S, P, R]: ...
def decorate2[
    S: HasAttr, **P, R
](func: Callable[Concatenate[S, P], R]) -> Callable[Concatenate[S, P], R]: ...

class A:
    my_attr: str

    @property
    @decorate1
    def prop1(self) -> int:
        return 1

    @property
    @decorate2
    def prop2(self) -> int:
        return 1

x = A()
reveal_type(x.prop1)
reveal_type(x.prop2)
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
